File: TechModule - Programming Fundamentals/07.DictionariesLambdaLinqEXERCISES/02.PhonebookUpgrade/PhonebookUpgrade.cs
using System;
using System.Collections.Generic;

namespace Phonebook
{
    class Phonebook
    {
        static void Main(string[] args)
        {
            SortedDictionary<string, string> phonebook = new SortedDictionary<string, string>();

            while (true)
            {
                string input = Console.ReadLine();
                if (input == "END")
                {
                    break;
                }

                string[] tokens = input.Split();
                string command = tokens[0];

                switch (command)
                {
                    case "A":
                        string name = tokens[1];
                        string phone = tokens[2];
                        if (phonebook.ContainsKey(name))
                        {
                            phonebook[name] = phone;
                        }
                        else
                        {
                            phonebook.Add(name, phone);
                        }
                        break;
                    case "S":
                        string searchedName = tokens[1];
                        if (phonebook.ContainsKey(searchedName))
                        {
                            string foundNumber = phonebook[searchedName];
                            Console.WriteLine($"{searchedName} -> {foundNumber}");
                        }
                        else
                        {
                            Console.WriteLine($"Contact {searchedName} does not exist.");
                        }
                        break;
                    case "ListAll":
                        foreach (var contact in phonebook)
                        {
                            Console.WriteLine($"{contact.Key} -> {contact.Value}");
                        }
                        break;
                }

            }
        }
    }
}

File: Software Technologies/09.JavaBasics/src/PhonebookUpgrade.java
import java.util.Scanner;
import java.util.TreeMap;

public class PhonebookUpgrade {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();

        TreeMap<String, String> phonebook = new TreeMap<>();

        while (!input.equals("END")) {
            String[] tokens = input.split(" ");
            String command = tokens[0];

            switch (command) {
                case "A":
                    String name = tokens[1];
                    String number = tokens[2];
                    phonebook.put(name, number);
                    break;
                case "S":
                    String contact = tokens[1];
                    if (!phonebook.containsKey(contact)) {
                        System.out.printf("Contact %s does not exist.%n", contact);
                    } else {
                        System.out.printf("%s -> %s%n", contact, phonebook.get(contact));
                    }
                    break;
                case "ListAll":
                    for (String key : phonebook.keySet()) {
                        System.out.printf("%s -> %s%n", key, phonebook.get(key));
                    }
                    break;
            }
            input = scanner.nextLine();
        }
    }
}
